src/core/init/create_eme_manager.ts

```diff
@@ -41,25 +46,28 @@ export default function createEMEManager(
   mediaElement : HTMLMediaElement,
   keySystems : IKeySystemOption[]
 ) : Observable<IEMEManagerEvent> {
+
+  // emit disabled event and handle unexpected encrypted event
+  function handleDisabledEME(logMessage: string): Observable<IEMEInitEvent> {
+    return observableConcat(
+      observableOf({ type: "eme-disabled" as "eme-disabled" }),
+      onEncrypted$(mediaElement).pipe(map(() => {
+        log.error(logMessage);
+        throw new EncryptedMediaError("MEDIA_IS_ENCRYPTED_ERROR", null, true);
```

Comment:
:O you make me think... Why not giving an error with that message as the "reason" of the EncryptedMediaError (second argument)?

That would be nice

Reply:
Still stands